### DISCUSSION TITLE
How to Revert a Merged PR

### DISCUSSION THREAD

**Discussion by jackfurr** [PyGithub/PyGithub#2177] [Q&A]
I can not figure out how to revert a merged PR.  can get information about the PR by calling `repo.get_pull(pr_number)`, but then what.  I could not find a way to revert a merged PR.

**Comment by bassil-overjet** [comment]
Hi @jackfurr - Did you find a way to do this? I'd also like to be able to revert merged PRs, but in my use case it isn't necessarily safe to create a new pr with the second to last commit in the commit tree.

**Reply by jackfurr** [reply]
I did not find a way to do this.  In the end, we decided that the revert process was not to be automated by Python.

**Reply by ltagliamonte-dd** [reply]
i'm in the same boat, @bassil-overjet did you find anything ?

**Reply by bassil-overjet** [reply]
@ltagliamonte-dd I didn't find a specific way to `pr.revert(pr_number)` but the operation is possible depending on your use case.
For our use case, we are only ever creating or updating one file at a time in the PRs that we automated, so we would "revert" a PR by creating/approving/merging a new PR with the operations that would have been performed by a revert. 

- If we want to revert a file that was created, we would get the contents of the file with its path in the repo, delete the file (`repo.delete(contents.path, **args)`), and create a new PR,
- If we wanted to revert a file that was updated, we would get the contents of the file, modify the contents of the file such that it was returned to its previous state, update the contents of the file with the "reverted" contents,  and create a new PR.

If in your use case you're reverting more than one file, I suppose our solution can be refactored to fit that use case by:

- enumerating all of the files that were created/updated/deleted during the PR but you now want to revert,
- for each file:
   - (1) - determining whether you will delete (revert a create), update (revert an update), or create (revert a delete),
   - perform the operation in (1)
- create a PR

**Comment by msummers42** [comment]
Interestingly, there is now an API for this https://github.blog/changelog/2023-01-27-api-for-reverting-a-pull-request/

**Reply by ltagliamonte-dd** [reply]
That's nice, they must have been added the feature, i remember speaking with the support about it.
